### PR TITLE
Initial conversion to a package

### PR DESCRIPTION
**Pull request by bakins**
Very "brute-force" attempt.  Bare minimum to convert to a package.  Avoided rearranging any code.  Runs and passes `go test`, but not much testing beyond that.  Consider this a conversation starter.

There were a few things I would like to do along with this, but I wanted to avoid doing too much in an initial attempt.

### CODE REVIEW

**Comment by aledbf** [comment]
@kelseyhightower any update on this?
Thanks

**Comment by bilalaslam** [comment]
@kelseyhightower any update on this?

**Comment by bacongobbler** [comment]
Initial thoughts on this looks great!

I'd prefer if we moved `/cmd/confd/confd.go` to `/confd/confd.go` for consistency with other projects like [fleetctl](https://github.com/coreos/fleet/tree/master/fleetctl) and [etcdctl](https://github.com/coreos/etcd/tree/master/etcdctl). Otherwise it looks good as a first pass which we can further improve upon in the future.

Would you also mind rebasing this off master, please?

**Comment by scalp42** [comment]
@bakins any chance to rebase this please?

**Comment by bakins** [comment]
Sorry, been busy. Will try tonight or in morning.

**Comment by bakins** [comment]
I did a "brute force rebase" with current master and squashed it.  Did very little testing, TBH.

File: build [hunk 1]
```diff
@@ -4,9 +4,8 @@ set -e
 
 GO="godep go"
 
-ORG_PATH="github.com/kelseyhightower"
-REPO_PATH="${ORG_PATH}/confd"
-
 echo "Building confd..."
+OLD=`pwd`
+cd confd
```
Comment (bacongobbler) [inline build]: This assumes that you're one directory below `confd`. I think it'd be better to use `CONFD_ROOT=$(cd $(dirname $0); pwd)` which will give you the absolute path to this directory.

Comment (bakins) [inline build]: Good catch. Thanks.  Fixed.

**Comment by bacongobbler** [comment]
That's okay. That's what [the](https://github.com/kelseyhightower/confd/pull/230) [integration](https://github.com/kelseyhightower/confd/pull/228) [tests](https://github.com/kelseyhightower/confd/pull/227) are for!

LGTM. ping @kelseyhightower for review

**Comment by bacongobbler** [comment]
note that this also transparently closes #245 as well as #208

**Comment by mboersma** [comment]
+1

**Comment by kelseyhightower** [comment]
@bakins Please rebase as I want to get it merged this week.

**Comment by bakins** [comment]
Sorry about the delay. Have had a busy couple of weeks.  With the recent changes, it may be easier for me to "hand patch" the changes to current master.  Have it on my board to get done today (Ha!).

**Comment by bakins** [comment]
Okay - attempt 2 done

**Comment by technosophos** [comment]
What's the status on this? Any update?

**Comment by bakins** [comment]
ENOTIME - whenever I get a few minutes to look at it, I need to do a full rebase.

**Comment by kelseyhightower** [comment]
Lets focus on cleaning up confd and getting 1.0 out the door. Then we can come back to this. I'll follow up with @bakins on this because I think this would be something useful for building on top of.

**Pull request closed without merging**